src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,155 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(tail, value, null);
+        if (size == 0) {
+            head = node;
+            tail = head;
+        }
+        tail.next = node;
+        tail = node;
+        size++;
+
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAddIndex(index);
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            addByIndexZero(value);
+        } else {
+            addByIndexDefault(index, value);
+        }
+
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return findByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = findByIndex(index);
+        T oldValue = node.item;
+        node.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> newNode = findByIndex(index);
+        T removeValue = newNode.item;
+        unLink(newNode);
+        size--;
+        return removeValue;
```

Comment:
```suggestion
        return unlink(findByIndex(index));
```
make unlink return T node instead of it being void and this method will be as simple as: